node/node.go

```diff
@@ -462,10 +473,55 @@ func (node *AlgorandFullNode) Ledger() *data.Ledger {
 	return node.ledger
 }
 
+// writeDevmodeBlock generates a new block for a devmode, and write it to the ledger.
+func (node *AlgorandFullNode) writeDevmodeBlock(lastRound basics.Round) (err error) {
+	// get the last block
+	blk, err := node.ledger.Block(lastRound)
+	if err != nil {
+		return
+	}
+
+	// modify the block blk round number, to emulate into-the-future block. The OnNewBlock doesn't
+	// really care about the rest of this data structure anyway, so it would be safe to do.
+	blk.BlockHeader.Round++
+
+	// calling OnNewBlock would trigger recomputation of the txn pool for round blk+1.
+	node.transactionPool.OnNewBlock(blk, ledgercore.StateDelta{})
+
+	// The above was already pregenerating the entire block,
+	// so there won't be any waiting on this call.
+	var vb *ledger.ValidatedBlock
+	vb, err = node.transactionPool.AssembleBlock(lastRound+1, time.Now().Add(config.ProposalAssemblyTime))
+	if err != nil || vb == nil {
```

Comment:
nit: maybe call it `NewDevModeBlock` and move to transaction pool in order to make easier to understand (well, subjective).
In this case `BroadcastSignedTxGroup` would call 3 funcs: ledger.Block(), pool.NewDevModeBlock(), ledger. AddValidatedBlock().